### ISSUE TITLE
Set like button for combination setter number?

### ISSUE THREAD

**Issue by oleg68** [GrandOrgue/grandorgue#1237]
### Discussed in https://github.com/GrandOrgue/grandorgue/discussions/1236

<div type='discussions-op-text'>

<sup>Originally posted by **Fl0r14n** November  1, 2022</sup>
I am wondering if there is a possibility of a button that will initiate a combination setter change state that will wait for 3 digits.
Currently to set combination 130, I have to press +100,  3 x +10 and _0. What I would like is a button (let's call it N) that when pressed will wait for me to add _1,  then _3, then _0 to jump to combination 130. For 024 will be N + _0 + _2 + _4

The existing functionality, for when I am on 130 and want to jump to 133, in which I just press _3, is great.

If I press N then press just 2 digits let's say, it will stay in a suspended state until I press the last digit. This is useful for when during a play, I have to switch from 130 back to let's say 126, without pressing 4 x < which creates side effects with registration (I know it's an edge case and why not duplicate the registration to 131 and just do > the whole piece)

That will help me a lot with my organ where I have 0 - 9 < > T C under the first manual, so that I don't use the keyboard

</div>

**Comment by oleg68** [comment]
@larspalo Do you agree with this enchancement?

**Comment by larspalo** [comment]
I'm not against the idea as such. There are indeed different ways pipe organ combination system works in this regard. Some wait until the user confirms the exact number entry (usually with another "enter" button) and then jumps to the desired combination number while others do indeed require that the user change "number bank range" first to jump to another number range of combinations.

The suggested way of a suspend button though is something I've not encountered - that's not to say that it might not exist...

On a side note - it's possible in the GUI from the toolbar to just enter a setter combination number in the box next to the <> button to jump to any combination.

My first thought about this would otherwise be to introduce a "flag" for selecting setter combination entry mode - for instance "direct" or "confirm" so that different behaviors could be simulated. A type of "enter" button would then also be needed to actually confirm the number entry if that mode is selected (confirm mode would actually disable the direct selection by the buttons for single digits ( __X ).

**Comment by larspalo** [comment]
@oleg68 Yes, I do understand the use case to select the setter combination at will with MIDI signals from a console. I only really question the method suggested, not the end result or the feature as such.

A delayed entry button would most likely be fairly easy to implement, but I've never seen anything like that on a real organ.

What I do have seen (and how the combination system of Bureå church main organ currently works) is that you have to enter the combination number by the physical number keys first and then push another enter button to confirm and activate the entered combination number.

**Comment by oleg68** [comment]
@larspalo I'd add a single `N` button that would activate the enter mode.

The `enter` would be made or after ertering three number buttons or after pressing `N` again. The another button `GC` would cancel the enter mode without changing the combination.

**Comment by larspalo** [comment]
@oleg68 please, don't use GC as a name for such an operation (that indicate "general cancel" instead). Why not use that N button with state? Just for toggling on/off the mode? And instead have an "Enter" or "Activate" next to it for usage when less than three digits are entered.

**Comment by oleg68** [comment]
@larspalo OK

three digits -> activate the combination
`N` again -> cancel

**Comment by bv74** [comment]
Hi, 
is functionality to set,  with MIDI event, the GUI field «combination number» implemented in GrandOrgue ? 
I'm beginner and havn't found that possibility in list of MIDI events.
Thank you for your help.

**Comment by efwalcker** [comment]
Done. https://github.com/GrandOrgue/grandorgue/issues/2430 
Thank you!
